ecs-agent/api/resource/ebs_discovery.go

```diff
@@ -44,21 +44,20 @@ func NewDiscoveryClient(ctx context.Context) *EBSDiscoveryClient {
 }
 
 // ScanEBSVolumes will iterate through the entire list of provided EBS volume attachments within the agent state and checks if it's attached on the host.
-func ScanEBSVolumes[T GenericEBSAttachmentObject](pendingAttachments map[string]T, dc EBSDiscovery) []string {
-	var err error
-	var foundVolumes []string
+func ScanEBSVolumes[T GenericEBSAttachmentObject](pendingAttachments map[string]T, dc EBSDiscovery) map[string]string {
+	foundVolumes := make(map[string]string)
 	for key, ebs := range pendingAttachments {
 		volumeId := strings.TrimPrefix(key, ebsResourceKeyPrefix)
-		deviceName := ebs.GetAttachmentProperties(DeviceName)
-		err = dc.ConfirmEBSVolumeIsAttached(deviceName, volumeId)
+		deviceName := ebs.GetAttachmentProperties(DeviceNameKey)
+		actualDeviceName, err := dc.ConfirmEBSVolumeIsAttached(deviceName, volumeId)
```

Comment:
same goes for the comment in `ConfirmEBSVolumeIsAttached` be clear that it returns the found device name -- otherwise call it something like `confirmAttachedAndGetDeviceName` or something.